src/injecting/injector.rs

```diff
@@ -106,20 +106,32 @@ impl FunctionMapper for Module {
         // actual index of the given function in own functions list.
 
         // Find the global function index
-        let global_function_index = self.get_global_function_index(function_name).unwrap();
+        let global_function_index = match self.get_global_function_index(function_name) {
+            Ok(global_function_index) => global_function_index,
+            Err(err) => return Err(err),
+        };
 
         // Count the number of imported functions listed by the module
-        let import_section_len: usize = self.get_import_section_len().unwrap();
+        let import_section_len: usize = match self.get_import_section_len() {
+            Ok(import_section_len) => import_section_len,
+            Err(err) => return Err(err),
+        };
 
-        let malloc_index = self.get_malloc_index().unwrap();
+        // get the global function index of the dynamic memory allocation method
+        let malloc_index = match self.get_malloc_index() {
+            Ok(malloc_index) => malloc_index,
+            Err(err) => return Err(err),
+        };
```

Comment:
This is redundant, you can use the `?` shorthand:

```rust
// Get the global function index of the dynamic memory allocation method
let malloc_index = self.get_malloc_index()?;
```

(Applies to the whole function)